src/scaladoc/scala/tools/nsc/doc/model/MemberLookup.scala

```diff
@@ -127,16 +127,13 @@ trait MemberLookup extends base.MemberLookupBase {
     }
   }
 
-  lazy val javaVersion: Int = {
-    var version = System.getProperty("java.version")
-    (if (version.startsWith("1.")) {
-      version.substring(2, 3)
+  lazy val javaVersion: Int =
+    System.getProperty("java.specification.version").split('.').take(2).map(_.toIntOption) match {
```

Comment:
`.split('.')` is `augmentString`, `split("\\.")` is java which notices it's a split on literal char, not a pattern. Just to footnote, as I always have to look it up.